lib/trip_dispatcher.rb

```diff
@@ -68,34 +68,72 @@ def load_trips
       trip_data = CSV.open('support/trips.csv', 'r', headers: true, header_converters: :symbol)
 
       trip_data.each do |raw_trip|
-        driver = find_driver(raw_trip[:driver_id].to_i)
+        driver = find_driver(raw_trip[:driver_id].to_i) # getting driver id from array, and putting in method called find driver
+        # find_driver method searches the driver csv for matching driver
         passenger = find_passenger(raw_trip[:passenger_id].to_i)
 
+
         parsed_trip = {
           id: raw_trip[:id].to_i,
-          driver: driver,
+          driver: driver, # driver symbol is set to the driver variable defined above,
+          # which is calling find_driver method through an instance variable (containing method load drivers)
           passenger: passenger,
-          start_time: raw_trip[:start_time],
-          end_time: raw_trip[:end_time],
+          start_time: Time.parse(raw_trip[:start_time]),
+          end_time: Time.parse(raw_trip[:end_time]),
           cost: raw_trip[:cost].to_f,
           rating: raw_trip[:rating].to_i
         }
 
         trip = Trip.new(parsed_trip)
+
+        # Set up the relations
         driver.add_trip(trip)
         passenger.add_trip(trip)
         trips << trip
       end
 
-      trips
+      return trips
     end
 
-    private
+    def request_trip(passenger_id)
+      # helper method for in_progress
+      trip_id = (trips.last.id) + 1
+      available_driver = @drivers.find {|driver| driver.status == :AVAILABLE}
+      start_time =  Time.now
+      # if driver is available then return their id # && make them UNAVAILABLE
+      passenger = find_passenger(passenger_id)
+
+
+      data = {
+        id: trip_id,
+        driver: available_driver,
+        passenger: passenger,
+        start_time: start_time,
+        end_time: nil,
+        cost: nil,
+        rating: nil,
+        }
+
+        trip = Trip.new(data)
+        available_driver.add_trip(trip)
+        available_driver.status = :UNAVAILABLE
+        passenger.add_trip(trip)
+
+        @trips << trip # trip added to instance variable @trips
+
+        return data # nilclass being returned for the trip rating
+      end
```

Comment:
Here you should be returning the instance of `Trip` in the `trip` variable, not the hash stored in `data`. They have all the same info, but the `Trip` is bundled into a neat package with methods you can call.